### ISSUE TITLE
Pinning the `Replayer` to a react div element

### ISSUE THREAD

**Issue by jay-khatri** [rrweb-io/rrweb#289]
Hi there. I'm using the `Replayer` in a react app and I'd like to render it within a component. My code looks something like this:

```
function AppInternal() {
	const { loading, error, data = [] } = useFetch(
		"http://localhost:8082/get-events?",
		{},
		[]
	);
	useEffect(() => {
		if (data && data.events && data.events.length > 1) {
			const replayer = new Replayer(data.events);
			replayer.wrapper = player;
			replayer.play();
		}
	}, [data]);
	if (loading) {
		return <p>loading...</p>;
	}
	if (error) {
		return <p>{error.toString()}</p>;
	}
	return <div id="player"/>;
}
```

This seems to be working, however I'd like to change the parent dom element of the player to be "#player". What's the easiest way to do this? Also, is there a better way to manage the state of the `Replayer` such that I can call the api methods easily?

Thanks

**Comment by jay-khatri** [comment]
I was able to get this working w/:

```
	const [replayer, setReplayer] = useState(undefined);
...
...
...
		if (data && data.events && data.events.length > 1) {
			setReplayer(
				new Replayer(data.events, {
					root: document.getElementById("player"),
					insertStyleRules: ["iframe {height: 400px;}"]
				})
			);
		}
```

I need to change the styles of the `iframe` now. Note that this is `<iframe height=<something>></iframe>`, not just a simple css attribute. Any thoughts there?

**Comment by Yuyz0112** [comment]
I think you can customize the iframe as a normal DOM. Are you facing any issue?

**Comment by Yuyz0112** [comment]
I think you can customize the iframe as a normal DOM. Are you facing any issue?

**Comment by jay-khatri** [comment]
Yes, got it working, thanks! Another question: the coordinates of the cursor that's overlayed on the iframe is slightly incorrect. What's the best way to go about debugging this? Is it based on the coordinates of `replayer-wrapper`?

**Comment by kanhegaonkarsaurabh** [comment]
@jay-khatri The mouse overlayed on the iframe is a sibling div element to the iframe (code pointer [here](https://github.com/rrweb-io/rrweb/blob/be4cccd6e373c1a6702130a21a1d4557f2944101/src/replay/index.ts#L287))
I'd start debugging this by looking at the CSS applied to this mouse div element in the styles [here](https://github.com/rrweb-io/rrweb/blob/be4cccd6e373c1a6702130a21a1d4557f2944101/src/replay/styles/style.css#L4) and how it is positioned relative to the wrapper div and the iframe.

**Comment by Yuyz0112** [comment]
Maybe related to some scroll things? The coordinates were recorded based on the window and replayed based on the iframe.
